pkg/web_app/test/widget/downloads_chart/downloads_chart_test.dart

```diff
@@ -78,4 +78,78 @@ void main() {
       expect(w3[i], [0, 0, 0, 0, 0, 0]);
     }
   });
+
+  group('closestPointOnLine tests', () {
+    test('point on the line', () {
+      final lineStart = (0, 0);
+      final lineEnd = (10, 10);
+      final point = (5, 5);
+      final closest = closestPointOnLine(lineStart, lineEnd, point);
+      expect(closest, (5.0, 5.0));
+    });
+
+    test('point before the line', () {
```

Comment:
If you want to test the behavior outside the valid input range, then please check both before and after. You could also skip this test entirely (invalid input).

Reply:
The next test is checking the after....